commit ca758b147e75f4b564225065d70b6526477185ce
Author: Mel Gorman <mgorman@techsingularity.net>
Date:   Thu Jan 23 22:11:15 2025 +0000

    fortify: Move FORTIFY_SOURCE under 'Kernel hardening options'
    
    FORTIFY_SOURCE is a hardening option both at build and runtime. Move
    it under 'Kernel hardening options'.
    
    Signed-off-by: Mel Gorman <mgorman@techsingularity.net>
    Acked-by: Paul Moore <paul@paul-moore.com>
    Link: https://lore.kernel.org/r/20250123221115.19722-5-mgorman@techsingularity.net
    Signed-off-by: Kees Cook <kees@kernel.org>

diff --git a/security/Kconfig.hardening b/security/Kconfig.hardening
index 45748379f896..23ffb0d7c845 100644
--- a/security/Kconfig.hardening
+++ b/security/Kconfig.hardening
@@ -282,6 +282,15 @@ endmenu
 
 menu "Bounds checking"
 
+config FORTIFY_SOURCE
+	bool "Harden common str/mem functions against buffer overflows"
+	depends on ARCH_HAS_FORTIFY_SOURCE
+	# https://github.com/llvm/llvm-project/issues/53645
+	depends on !CC_IS_CLANG || !X86_32
+	help
+	  Detect overflows of buffers in common string and memory functions
+	  where the compiler can determine and validate the buffer sizes.
+
 config HARDENED_USERCOPY
 	bool "Harden memory copies between kernel and userspace"
 	imply STRICT_DEVMEM